lgc/patch/RegisterMetadataBuilder.cpp

```diff
@@ -0,0 +1,1380 @@
+  }
+
+  const unsigned ldsSizeDwordGranularityShift =
+      m_pipelineState->getTargetInfo().getGpuProperty().ldsSizeDwordGranularityShift;
+  const unsigned ldsSizeDwordGranularity = 1u << ldsSizeDwordGranularityShift;
+
+  unsigned ldsSizeInDwords = calcFactor.gsOnChipLdsSize;
+#if VKI_RAY_TRACING
+  ldsSizeInDwords += calcFactor.rayQueryLdsStackSize;
+#endif
+  ldsSizeInDwords = alignTo(ldsSizeInDwords, ldsSizeDwordGranularity);
+
+  auto hwShaderNode = getHwShaderNode(Util::Abi::HardwareStage::Gs);
+  hwShaderNode[Util::Abi::HardwareStageMetadataKey::LdsSize] = ldsSizeInDwords * 4;
+  if (!m_hasMesh)
+    setEsGsLdsSize(calcFactor.esGsLdsSize * 4);
+}
+
+// =====================================================================================================================
+// Builds register configuration for hardware vertex shader.
+void RegisterMetadataBuilder::buildHwVsRegisters() {
+  assert(m_hasVs || m_hasTes || m_pipelineState->hasShaderStage(ShaderStageCopyShader));
+  assert(m_gfxIp.major <= 10);
+  ShaderStage shaderStage = ShaderStageVertex;
+  if (m_pipelineState->hasShaderStage(ShaderStageCopyShader))
+    shaderStage = ShaderStageCopyShader;
+  else if (m_hasTes)
+    shaderStage = ShaderStageTessEval;
+
+  const auto resUsage = m_pipelineState->getShaderResourceUsage(shaderStage);
+  const auto &builtInUsage = resUsage->builtInUsage;
+
+  const auto &xfbStrides = m_pipelineState->getXfbBufferStrides();
+  const auto &streamXfbBuffers = m_pipelineState->getStreamXfbBuffers();
+  const bool enableXfb = m_pipelineState->enableXfb();
+
+  // VGT_STRMOUT_CONFIG
+  auto vgtStrmoutConfig = getGraphicsRegNode()[Util::Abi::GraphicsRegisterMetadataKey::VgtStrmoutConfig].getMap(true);
+  vgtStrmoutConfig[Util::Abi::VgtStrmoutConfigMetadataKey::Streamout_0En] = streamXfbBuffers[0] > 0;
+  vgtStrmoutConfig[Util::Abi::VgtStrmoutConfigMetadataKey::Streamout_1En] = streamXfbBuffers[1] > 0;
+  vgtStrmoutConfig[Util::Abi::VgtStrmoutConfigMetadataKey::Streamout_2En] = streamXfbBuffers[2] > 0;
+  vgtStrmoutConfig[Util::Abi::VgtStrmoutConfigMetadataKey::Streamout_3En] = streamXfbBuffers[3] > 0;
+  if (shaderStage == ShaderStageCopyShader)
+    vgtStrmoutConfig[Util::Abi::VgtStrmoutConfigMetadataKey::RastStream] = resUsage->inOutUsage.gs.rasterStream;
+
+  // Set some field of SPI_SHADER_PGM_RSRC2_VS
+  getGraphicsRegNode()[Util::Abi::GraphicsRegisterMetadataKey::VsSoEn] = enableXfb;
+  getGraphicsRegNode()[Util::Abi::GraphicsRegisterMetadataKey::VsSoBase0En] = xfbStrides[0] > 0;
+  getGraphicsRegNode()[Util::Abi::GraphicsRegisterMetadataKey::VsSoBase1En] = xfbStrides[1] > 0;
+  getGraphicsRegNode()[Util::Abi::GraphicsRegisterMetadataKey::VsSoBase2En] = xfbStrides[2] > 0;
+  getGraphicsRegNode()[Util::Abi::GraphicsRegisterMetadataKey::VsSoBase3En] = xfbStrides[3] > 0;
+
+  // VGT_STRMOUT_VTX_STRIDE_*
+  getGraphicsRegNode()[Util::Abi::GraphicsRegisterMetadataKey::VgtStrmoutVtxStride0] = xfbStrides[0] / sizeof(unsigned);
```

Comment:
These four lines give errors "use of overloaded operator= is ambiguous" for me. The right side is unsigned long. Probably need to manually cast to unsigned.

Reply:
Thank you for pointing out. I will fix them.

Reply:
Fix in PR #2264